src/styles/calendar.scss

```diff
@@ -0,0 +1,48 @@
+@import "./_variables.scss";
+
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  padding: 10px;
+  gap: 1px;
+
+  @each $day, $margin in $days {
+    &-start-day--#{$day} &__day:first-child {
+      margin-left: $margin;
+    }
+  }
+
+  &__day {
+    @include square($cellSize);
+
+    font-family: Arial, sans-serif;
+    font-size: 30px;
+
+    background-color: #eee;
+    border: 1px solid black;
+    cursor: pointer;
+
+    transition: background-color 0.5s, transform 0.5s;
+
+    @for $day from 1 through 31 {
+      &:nth-child(#{$day}):before {
+        content: "#{$day}";
+        display: flex;
+        height: 100%;
+        justify-content: center;
+        align-items: center;
```

Comment:
add all these styles outside the loop, leave in the loop styles that depend only on the iteration of the loop